ext/cl_khr_command_buffer.asciidoc

```diff
@@ -457,8 +458,14 @@ Add to *Table 5*, _Device Queries_, of section 4.2:
   {CL_COMMAND_BUFFER_CAPABILITY_SIMULTANEOUS_USE_KHR} Device supports the
   command-buffers having a <<pending_count, Pending Count>> that exceeds 1.
 
-  {CL_COMMAND_BUFFER_CAPABILITY_OUT_OF_ORDER_KHR} Device supports the ability
-  to record command-buffers to out-of-order command-queues.
+| {CL_DEVICE_COMMAND_BUFFER_SUPPORTED_QUEUE_PROPERTIES_KHR}
+| {cl_command_queue_properties_TYPE}
+| Bitmask of the supported properties with which a command-queue may be created
+  to allow a command-buffer to be executed on it. It is invalid for a
+  command-queue to be created with a property not reported and still be
+  compatible with command-buffer execution.
```

Comment:
Are the queue properties for executing a command buffer mandated to be the same as the queue properties while recording the command buffer.

Reply:
Yes we currently have this [wording](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_Ext.html#compatible) in the spec about being *compatible*
> Command-buffers are created using an ordered list of command-queues that commands are recorded to and execute on by default. These command-queues can be replaced on command-buffer enqueue with different command-queues, provided for each element in the replacement list the substitute command-queue is compatible with the command-queue used on command-buffer creation. Where a **compatible** command-queue is defined as a command-queue with identical properties targeting the same device and in the same OpenCL context.

And then the actual error condition to `clEnqueueCommandBufferKHR` is
>CL_INCOMPATIBLE_COMMAND_QUEUE_KHR if any element of queues is not compatible with the command-queue set on command_buffer creation at the same list index.


I've ignored that `clSetCommandQueueProperty` exists because its been depreciated since OpenCL 1.1, as that could change a command-queue's properties after creation. Maybe I should put a note about not using that.